src/Classes/ItemListControl.lua

```diff
@@ -42,9 +42,22 @@ local ItemListClass = newClass("ItemListControl", "ListControl", function(self,
 	end
 	self.controls.deleteUnused = new("ButtonControl", {"RIGHT",self.controls.deleteAll,"LEFT"}, -4, 0, 100, 18, "Delete Unused", function()
 		local delList = {}
+		local abyssalJewelsList = {}
+		local abyssalJewelsOwnedbyItemsList = {}
 		for _, itemId in pairs(self.list) do
-			if not itemsTab:GetEquippedSlotForItem(itemsTab.items[itemId]) and not self:FindSocketedJewel(itemId, false) then
-				t_insert(delList, itemId)
+			-- Abyssal jewels will not show up using FindSocketedJewel
+			-- Get a list of abyssal jewels as the list moves thru and deal with them last
+			local item = self.itemsTab.items[itemId]
+			if item.type == "Jewel" then
+				if self:FindAbyssalJewel(item) == "" and self:FindSocketedJewel(itemId, false) == "" then
```

Comment:
Returning empty strings for failed matches on ``FindAbyssalJewel`` and ``FindSocketedJewel`` is unnecessary and inconsistent with ``GetEquippedSlotForItem`` behavior when it fails to find a match.

Reply:
I can't see the difference but ok.
I've converted them all back to nil or string. and check all the usages of them to make sure they don't compare to a string

![image](https://user-images.githubusercontent.com/4302241/152519695-e28e983a-8105-4187-a4dc-8699db124b82.png)
Now the 'Used in' next to the Abyssal Jewel is on permanently. It's what happened when i copied your code first, so i went back to strings.

Maybe i've missed one.
X